shared/test/unit/mocks/mock_navigator_moz_settings.js

```diff
@@ -64,6 +64,11 @@
     var req = {
       addEventListener: function(name, cb) {
         req['on' + name] = cb;
+      },
+      then: function(cb, eb) {
+        // error-handler `eb`is never used, since the mock never fails.
+        return Promise.resolve(0).then(cb); // set resolve to 0.
+
```

Comment:
I'd really like to keep the functionality around "mSyncRepliesOnly". The idea is that a test developer could want to decide when the request is actually answered.

That's why I advised to use directly addEventListener, even if it was feeling more hacky :)

Tell me what you think